File: sdk/basin-java/src/main/java/io/basin/sdk/OAuthAuthorizeResult.java
package io.basin.sdk;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Response from {@code GET /auth/v1/oauth/:provider/authorize}.
 *
 * <p>Redirect the user's browser to {@link #redirectUrl}. The server's callback
 * handler ({@code GET /auth/v1/oauth/:provider/callback}) exchanges the code and
 * issues a Basin token pair. The SDK cannot perform the browser redirect itself.
 */
public final class OAuthAuthorizeResult {

    /** Full provider authorize URL — redirect the browser here. */
    @JsonProperty("redirect_url")
    public final String redirectUrl;

    /** CSRF state value included in {@link #redirectUrl}; echoed on callback. */
    @JsonProperty("state")
    public final String state;

    public OAuthAuthorizeResult(
            @JsonProperty("redirect_url") String redirectUrl,
            @JsonProperty("state") String state) {
        this.redirectUrl = redirectUrl;
        this.state = state;
    }
}

File: sdks/php/src/Basin/Types/OAuthAuthorizeResult.php
<?php

declare(strict_types=1);

namespace Basin\Types;

/**
 * GET /auth/v1/oauth/:provider/authorize response.
 *
 * Redirect the user's browser to $redirectUrl to begin the provider OAuth
 * flow. The server's callback (GET /auth/v1/oauth/:provider/callback) will
 * exchange the code and issue Basin JWT + refresh tokens.
 */
final class OAuthAuthorizeResult
{
    public function __construct(
        /** Full provider authorize URL — redirect the browser here. */
        public readonly string $redirectUrl,
        /** CSRF state value included in redirect_url; echoed back on callback. */
        public readonly string $state,
    ) {
    }

    /** @param array<string,mixed> $data */
    public static function fromArray(array $data): self
    {
        return new self(
            redirectUrl: (string) $data['redirect_url'],
            state: (string) $data['state'],
        );
    }
}
